sechub-wrapper-owasp-zap/src/test/java/com/mercedesbenz/sechub/zapwrapper/config/ZapScanContextFactoryTest.java

```diff
@@ -318,122 +297,85 @@ void commandline_settings_null_throws_zap_wrapper_runtime_exception() {
+        CommandLineSettings settings = createSettingsMock();
+        when(envVariableReader.readAsString(ZAP_DEACTIVATED_RULE_REFERENCES)).thenReturn(null);
 
         /* execute */
         ZapScanContext result = factoryToTest.create(settings);
-        DeactivatedRuleReferences deactivatedRuleReferences = result.getDeactivatedRuleReferences();
+        List<String> zapRuleIDsToActivate = result.getZapRuleIDsToDeactivate();
 
         /* test */
-        verify(ruleProvider).fetchDeactivatedRuleReferences(any());
-        assertNotNull(deactivatedRuleReferences);
-        assertNotNull(deactivatedRuleReferences.getDeactivatedRuleReferences());
-        assertEquals(2, deactivatedRuleReferences.getDeactivatedRuleReferences().size());
-
+        assertNotNull(zapRuleIDsToActivate);
+        assertTrue(zapRuleIDsToActivate.isEmpty());
     }
 
-    @ParameterizedTest
-    @NullSource
-    @ValueSource(strings = { "", "Cross-Site-Scripting-(DOM-Based)-40026,Timestamp-Disclosure-10096",
-            "Cross-Site-Scripting-(DOM-Based)-40026,Timestamp-Disclosure-10096,Cross-Domain-Misconfiguration-10098" })
-    void rules_to_deactivate_returned_by_env_variable_is_inside_result(String value) {
+    @Test
+    void rules_to_deactivate_returned_by_env_variable_is_inside_result() {
         /* prepare */
-        CommandLineSettings settings = createSettingsMockWithNecessaryPartsWithoutRuleFiles();
-        when(envVariableReader.readAsString(ZAP_DEACTIVATED_RULE_REFERENCES)).thenReturn(value);
-        when(ruleProvider.fetchDeactivatedRuleReferences(any())).thenReturn(new DeactivatedRuleReferences());
+        CommandLineSettings settings = createSettingsMock();
+        when(settings.getDeactivateRules()).thenReturn(Collections.emptyList());
 
-        String[] arrayToTestExpectedLength = {};
-        if (value != null) {
-            arrayToTestExpectedLength = value.split(",");
-        }
+        String ruleIds = "12345,321";
+        when(envVariableReader.readAsString(ZAP_DEACTIVATED_RULE_REFERENCES)).thenReturn(ruleIds);
+        List<String> expected = Arrays.asList(ruleIds.split(","));
 
         /* execute */
         ZapScanContext result = factoryToTest.create(settings);
-        DeactivatedRuleReferences deactivatedRuleReferences = result.getDeactivatedRuleReferences();
+        List<String> zapRuleIDsToActivate = result.getZapRuleIDsToDeactivate();
 
         /* test */
-        assertNotNull(deactivatedRuleReferences);
-        assertNotNull(deactivatedRuleReferences.getDeactivatedRuleReferences());
-        assertEquals(arrayToTestExpectedLength.length, deactivatedRuleReferences.getDeactivatedRuleReferences().size());
+        assertNotNull(zapRuleIDsToActivate);
+        assertEquals(expected, zapRuleIDsToActivate);
     }
 
     @ParameterizedTest
     @ValueSource(strings = { "", "Cross-Site-Scripting-(DOM-Based)-40026,Timestamp-Disclosure-10096",
```

Comment:
The ids here are strings - e.g. `Cross-Site-Scripting-(DOM-Based)-40026`.
But the ids from real ZAP would be "123,345" etc. 
Should we change the test here?